src/entry/aws-console.ts

```diff
@@ -206,6 +207,68 @@ function delay(ms: number) {
   return new Promise( resolve => setTimeout(resolve, ms) );
 }
 
+interface AccountMenuInfo {
+  accountId: string | null;
+  roleName: string | null;
+  ssoRoleName: string | null;
+  userType: 'iam' | 'sso';
+}
+
+const parseAccountMenu = (accountMenu: HTMLElement, prompts: string[]): AccountMenuInfo => {
+  // scrape values from the aws console account menu (top right)
+  // some pages are using an older version of the menu
+  // the old menu prompt includes the ': ', new one does not
+  // old iam: 'Currently active as: '
+  // old sso: 'Account ID: '
+  // new iam: 'Currently active as'
+  // new sso: 'Account ID'
+  const oldAccountPrompt = accountMenu.firstElementChild?.getElementsByTagName('span')[0].textContent || '';
+  
+  if (oldAccountPrompt === 'Currently active as: ') {
+    return {
+      userType: 'iam',
+      accountId: accountMenu.lastElementChild!.getElementsByTagName('span')[1].textContent!.replaceAll('-', ''),
+      roleName: accountMenu.firstElementChild!.getElementsByTagName('span')[1].textContent!,
+      ssoRoleName: null
+    };
+  }
+  
+  if (oldAccountPrompt.replace(': ', '') && prompts.includes(oldAccountPrompt.replace(': ', ''))) {
+    const accountId = accountMenu.firstElementChild!.getElementsByTagName('span')[1].textContent!.replaceAll('-', '');
+    const roleName = accountMenu.lastElementChild!.getAttribute('title')!;
+    return {
+      userType: 'sso',
+      accountId,
+      roleName,
+      ssoRoleName: ssoRoleName(roleName)
+    };
+  }
+  
+  const allDivs = Array.from(accountMenu.querySelectorAll('div'));
+  const accountIdDiv = allDivs.find(div => prompts.includes(div.textContent?.trim() ?? ''));
+  
+  if (accountIdDiv) {
+    const accountIdSpan = accountIdDiv.parentElement?.querySelector('div > span:last-of-type');
+    const accountId = accountIdSpan?.textContent?.replaceAll('-', '') || null;
+    const roleNameDiv = accountMenu.querySelector('div[title]');
+    const roleName = roleNameDiv?.getAttribute('title') || null;
+    
+    return {
+      userType: 'sso',
+      accountId,
+      roleName,
+      ssoRoleName: roleName ? ssoRoleName(roleName) : null
+    };
+  }
+  
+  return {
+    userType: 'iam',
+    accountId: accountMenu.getElementsByTagName('span')[7].textContent!.replaceAll('-', ''),
+    roleName: accountMenu.getElementsByTagName('span')[3].textContent!,
```

Comment:
I didn't had an IAM account, only SSO, so I'm not 100% sure I haven't broken something here for IAM.